src/common/core/metacore.js

```diff
@@ -222,12 +222,14 @@ define([
         };
 
         this.getValidPointerNames = function (node) {
-            var validNames = self.getPointerNames(getMetaNode(node)) || [],
+            var metaDefNode = getMetaNode(node),
+                validNames = self.getPointerNames(metaDefNode) || [],
                 i,
                 validPointerNames = [],
                 metaPointerNode, max;
+
             for (i = 0; i < validNames.length; i++) {
-                metaPointerNode = getMetaPointerNode(node, validNames[i]);
+                metaPointerNode = self.getChild(metaDefNode, CONSTANTS.META_POINTER_PREFIX + validNames[i]);
```

Comment:
Why replacing the getMetaPointerNode in some places and leave untouched in others?
Either the functions should be changed or it should be removed from everywhere.
(or just I am not seeing the distinction among these usages)